### DISCUSSION TITLE
Returning Tasks when possible instead of awaiting them and usage of .configureawait(false)

### DISCUSSION THREAD

**Discussion by robsiera** [coronabytes/dotnet-arangodb#63] [Ideas]
I would like your opinion on two topics, before I present you a pull request.

### Returning Tasks when possible instead of awaiting them

There are still places in your library where a function returns an awaited result value while it could just return the Task, as the return statement is the last line in that function.  Meaning this

```
        public async Task ImportAsync<T>(ArangoHandle database, string collection, IEnumerable<T> docs,
            bool complete = true,
            CancellationToken cancellationToken = default)
        {
            ... doing stuff ...
            return await SendAsync<ArangoVoid>(database, HttpMethod.Post, query,  docs,  cancellationToken: cancellationToken);
        }
```

can be changed to this

```
        public Task ImportAsync<T>(ArangoHandle database, string collection, IEnumerable<T> docs,
            bool complete = true,
            CancellationToken cancellationToken = default)
        {
            ... doing stuff ...
            return SendAsync<ArangoVoid>(database, HttpMethod.Post, query,  docs,  cancellationToken: cancellationToken);
        }
```

thus it will be the above code that will await the task. While now the task is awaited twice.

What is your opinion on this observation?

### usage of .configureawait(false)

At some point in time, a lot of .configureawait(false) statements where added to your library. But I have noticed that this isn't used throughout your code. Is this deliberate? Is there a rationale behind the current use? Sometimes Functions use .configureawait(false) while other, similar functions do not.

**Comment by coronabytes** [comment]
Hi,
yes a pull request for that is very welcome,

- configureawait(false) should be used everywhere if its (server) library code (i still don't get it why there isn't a global switch)
- returning tasks directly is also fine, some methods are alot of copypasta and some had additional awaits before they were removed and i just didn't return it directly even if i could
